### PR TITLE
HPM countries

### PR DESCRIPTION
**Pull request by cetvrtak**

### CODE REVIEW

File: EU4toV2/Source/Mappers/CountryMappings/CountryMappings.cpp [hunk 1]
```diff
@@ -278,7 +278,20 @@ std::optional<std::string> mappers::CountryMappings::getV2Tag(const std::string&
 	}
 
 	if (const auto& findIter = eu4TagToV2TagMap.find(eu4Tag); findIter != eu4TagToV2TagMap.end())
-		return findIter->second;
+	{
+		if (const auto& mappingIter = std::find_if(eu4TagToV2TagsRules.begin(), eu4TagToV2TagsRules.end(),
```
Comment (Zemurin) [inline EU4toV2/Source/Mappers/CountryMappings/CountryMappings.cpp:282]: When you overwrite v2 field, this becomes unnecessary.

File: EU4toV2/Source/Mappers/CountryMappings/CountryMapping.cpp [hunk 2]
```diff
@@ -17,6 +17,9 @@ void mappers::CountryMapping::registerKeys()
 	registerKeyword("v2", [this](const std::string& unused, std::istream& theStream) {
 		vic2Tag = commonItems::singleString(theStream).getString();
 	});
+	registerKeyword("hpm", [this](const std::string& unused, std::istream& theStream) {
```
Comment (Zemurin) [inline EU4toV2/Source/Mappers/CountryMappings/CountryMapping.cpp:20]: this is fine, but at the end of the constructor you should overwrite v2 field with hpm field if configuration:HPM is enabled.

File: EU4toV2/Source/Mappers/CountryMappings/CountryMapping.h [hunk 3]
```diff
@@ -13,6 +13,7 @@ class CountryMapping: commonItems::parser
 
 	[[nodiscard]] const auto& getEU4Tag() const { return eu4Tag; }
 	[[nodiscard]] const auto& getVic2Tag() const { return vic2Tag; }
+	[[nodiscard]] const auto& getHpmTag() const { return hpmTag; }
```
Comment (Zemurin) [inline EU4toV2/Source/Mappers/CountryMappings/CountryMapping.h:16]: shouldn't be needed.

**Review by Zemurin — CHANGES_REQUESTED** [review]

**Review by Idhrendur — APPROVED** [review]

File: EU4toV2/Source/Mappers/CountryMappings/CountryMapping.cpp (hunk 2 above)
Comment (cetvrtak) [reply EU4toV2/Source/Mappers/CountryMappings/CountryMapping.cpp:20]: Brilliant

**Review by Zemurin — APPROVED** [review]

**Pull request merged**